appinventor/blocklyeditor/src/replmgr.js

```diff
@@ -1608,36 +1552,15 @@ Blockly.ReplMgr.resendAssetsAndExtensions = function() {
 
 Blockly.ReplMgr.loadExtensions = function() {
     var rs = top.ReplState;
-    if (top.usewebrtc) {
-        // Need to trigger the loading of extensions here
-        rs.state = Blockly.ReplMgr.rsState.EXTENSIONS;
-        var extensionJson = JSON.stringify(top.AssetManager_getExtensions());
-        extensionJson = extensionJson.replace(/"/g, '\\"');
-        var yailstring = "(AssetFetcher:loadExtensions \"" +
-            extensionJson +
-            "\")";
-        console.log("Blockly.ReplMgr.loadExtensions(webrtc): Yail = " + yailstring);
-        this.putYail.putAsset(yailstring);
-    } else {
-        rs.state = Blockly.ReplMgr.rsState.EXTENSIONS;
-        var xmlhttp = goog.net.XmlHttp();
-        var encoder = new goog.Uri.QueryData();
-        encoder.add('extensions', JSON.stringify(top.AssetManager_getExtensions()));
-        xmlhttp.open('POST', rs.extensionurl, true);
-        xmlhttp.onreadystatechange = function() {
-            /* Older companions do not support ReplMgr providing an extension list. This check below
-             * handles older companions as well as new companions so that we don't break old clients.
-             */
-            if (xmlhttp.readyState === 4 && (this.status === 200 || this.status === 404 || !this.status)) {
-                rs.state = Blockly.ReplMgr.rsState.CONNECTED;
-                Blockly.mainWorkspace.fireChangeListener(new AI.Events.CompanionConnect());
-            } else if (xmlhttp.readyState === 4) {
-                rs.state = Blockly.ReplMgr.rsState.IDLE;
-                top.BlocklyPanel_indicateDisconnect();
-            }
-        };
-        xmlhttp.send(encoder.toString());
-    }
+    // Need to trigger the loading of extensions here
+    rs.state = Blockly.ReplMgr.rsState.EXTENSIONS;
+    var extensionJson = JSON.stringify(top.AssetManager_getExtensions());
+    extensionJson = extensionJson.replace(/"/g, '\\"');
```

Comment:
Not that I think it could happen (maybe a maliciously crafted components.json?), but if the extensions had `\"` in the string, this would make it `\\"` rather than `\\\"`. We may want to call `Blockly.Yail.quotifyForREPL` instead since it is more robust.